geoportailv3/static/js/search/searchdirective.js

```diff
@@ -0,0 +1,356 @@
+/**
+ * @fileoverview This file provides a "search" directive. This directive is
+ * used to insert a Search bar into a HTML page.
+ * Example:
+ *
+ * <app-search app-search-map="::mainCtrl.map"></app-search>
+ *
+ * Note the use of the one-time binding operator (::) in the map expression.
+ * One-time binding is used because we know the map is not going to change
+ * during the lifetime of the application.
+ *
+ */
+goog.provide('app.searchDirective');
+
+goog.require('app');
+goog.require('app.GetLayerForCatalogNode');
+goog.require('app.ShowLayerinfo');
+goog.require('app.Themes');
+goog.require('ngeo.CreateGeoJSONBloodhound');
+goog.require('ngeo.searchDirective');
+
+
+/**
+ * @return {angular.Directive} The Directive Object Definition
+ * @param {string} appSearchTemplateUrl
+ * @ngInject
+ */
+app.searchDirective = function(appSearchTemplateUrl) {
+  return {
+    restrict: 'E',
+    scope: {
+      'map': '=appSearchMap'
+    },
+    controller: 'AppSearchController',
+    controllerAs: 'ctrl',
+    bindToController: true,
+    templateUrl: appSearchTemplateUrl,
+    link:
+        /**
+         * @param {angular.Scope} scope Scope
+         * @param {angular.JQLite} element Element
+         * @param {angular.Attributes} attrs Atttributes
+         */
+        function(scope, element, attrs) {
+          // Empty the search field on focus and blur.
+          element.find('input').on('focus blur', function() {
+            $(this).val('');
+          });
+        }
+  };
+};
+
+
+app.module.directive('appSearch', app.searchDirective);
+
+
+
+/**
+ * @ngInject
+ * @constructor
+ * @param {angular.Scope} $scope Angular root scope.
+ * @param {app.Themes} appThemes Themes service.
+ * @param {angular.$compile} $compile Angular compile service.
+ * @param {ngeo.CreateGeoJSONBloodhound} ngeoCreateGeoJSONBloodhound The ngeo
+ *     create GeoJSON Bloodhound service.o
+ * @param {angularGettext.Catalog} gettextCatalog Gettext catalog
+ * @param {app.GetLayerForCatalogNode} appGetLayerForCatalogNode
+ * @param {app.ShowLayerinfo} appShowLayerinfo
+ * @param {string} searchServiceUrl
+ * @export
+ */
+app.SearchDirectiveController =
+    function($scope, appThemes, $compile,
+        ngeoCreateGeoJSONBloodhound, gettextCatalog,
+        appGetLayerForCatalogNode, appShowLayerinfo, searchServiceUrl) {
+  /**
+   * @type {Array}
+   */
+  this.layers = [];
```

Comment:
I think that you have a problem here. You declare the `layers` using the dot notation here (`this.layers`), and then refer to the array using the square bracket notation (`this['layers']`). You need to be consistent and use the same notation everywhere.

And for properties that are not used in Angular views/templates it's better to use the dot notation, so the compiler can rename the properties with shorter names.

So, in this case, you can probably use this:

```js
/**
 * @type {Array.<ol.layer.Layer>}
 * @private
 */
this.layers_ = [];
```

and always use `this.layers_` further down in the code to refer to the array.